3/typedefs.js

```diff
@@ -0,0 +1,45 @@
+const { gql } = require('apollo-server');
+
+const typeDefs = gql`
+
+  enum State {
+    DONE
+    UNDONE
+    IN_PROGRESS
+  }
+
+  type Todo {
+    id: ID!
+    message: String
+    state: State
+    assigneeId: ID
+  }
+
+  type Assignee {
+    id: ID!
+    name: String
+  }
+
+  type Query {
+    assignees(token: String!): [Assignee]
+    assignee(id: ID!): Assignee
+    todos: [Todo]
+    todo(id: ID!): Todo
+    todosForAssigneeId(assigneeId: ID!): [Todo]
+    countTodos: Int
+    countAssignees: Int
+    todosByState(state: State!): [Todo]
```

Comment:
I would merge a couple of these queries here and add a `filter` argument as described above :point_up: